src/App/Pages/Settings/LoginPasswordlessRequestsListViewModel.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Bit.App.Abstractions;
+using Bit.App.Resources;
+using Bit.Core.Abstractions;
+using Bit.Core.Models.Response;
+using Bit.Core.Utilities;
+using Xamarin.CommunityToolkit.ObjectModel;
+using Xamarin.Forms;
+
+namespace Bit.App.Pages
+{
+    public class LoginPasswordlessRequestsListViewModel : BaseViewModel
+    {
+        private readonly IAuthService _authService;
+        private readonly IStateService _stateService;
+        private readonly IDeviceActionService _deviceActionService;
+        private readonly IPlatformUtilsService _platformUtilsService;
+        private readonly ILogger _logger;
+        private ObservableRangeCollection<PasswordlessLoginResponse> _loginRequestsList;
+        private bool _isRefreshing;
+
+        public LoginPasswordlessRequestsListViewModel()
+        {
+            _authService = ServiceContainer.Resolve<IAuthService>();
+            _stateService = ServiceContainer.Resolve<IStateService>();
+            _deviceActionService = ServiceContainer.Resolve<IDeviceActionService>();
+            _platformUtilsService = ServiceContainer.Resolve<IPlatformUtilsService>();
+            _logger = ServiceContainer.Resolve<ILogger>();
+
+            PageTitle = AppResources.PendingLogInRequests;
+            LoginRequestsList = new ObservableRangeCollection<PasswordlessLoginResponse>();
+
+            AnswerRequestCommand = new AsyncCommand<PasswordlessLoginResponse>(PasswordlessLoginAsync,
+                onException: ex => HandleException(ex),
+                allowsMultipleExecutions: false);
+
+            DeclineAllRequestsCommand = new AsyncCommand(DeclineAllRequestsAsync,
+                onException: ex => HandleException(ex),
+                allowsMultipleExecutions: false);
+
+            RefreshCommand = new AsyncCommand(RefreshAsync,
+                onException: ex => HandleException(ex),
+                allowsMultipleExecutions: false);
+        }
+
+        public ICommand RefreshCommand { get; }
+
+        public AsyncCommand<PasswordlessLoginResponse> AnswerRequestCommand { get; }
+
+        public AsyncCommand DeclineAllRequestsCommand { get; }
+
+        public ObservableRangeCollection<PasswordlessLoginResponse> LoginRequestsList
+        {
+            get => _loginRequestsList;
+            set => SetProperty(ref _loginRequestsList, value);
+        }
+
+        public bool IsRefreshing
+        {
+            get => _isRefreshing;
+            set => SetProperty(ref _isRefreshing, value);
+        }
```

Comment:
After trying to change the code i found couple more bugs with `RefreshView`. One cannot use `AsyncCommands` or `CanExecute` functions because it messes the control of the variables in the `RefreshView`.
source: https://github.com/xamarin/Xamarin.Forms/issues/14350

Reply:
Made necessary changes in order to make it work.